mcs/class/System/System.Net/WebCompletionSource.cs

```diff
@@ -66,14 +72,14 @@ public void ThrowOnError ()
 			completion.Task.Result.Error?.Throw ();
 		}
 
-		public async Task<bool> WaitForCompletion (bool throwOnError)
+		public async Task<T> WaitForCompletion ()
 		{
 			var result = await completion.Task.ConfigureAwait (false);
 			if (result.State == State.Completed)
-				return true;
-			if (throwOnError)
-				result.Error.Throw ();
-			return false;
+				return (T)result.Argument;
+			// This will always throw once we get here.
+			result.Error.Throw ();
+			throw new InvalidOperationException ("Should never happen.");
```

Comment:
Why do you need this line?

Reply:
[`ExceptionDispatchInfo.Throw()`](https://msdn.microsoft.com/en-us/library/system.runtime.exceptionservices.exceptiondispatchinfo.throw(v=vs.110).aspx) always throws an exception, but its return type if `void`.
We will never actually reach line 82 here, but the compiler would still complain about it.
Any better suggestions how to deal with this?  I could surely use `return default (T)` here, just thought throwing an exception is the cleaner way of dealing with a line that can never be reached.

Reply:
I thought as this is async Task based method it should just pass the exception via task e.g. Task<T>.FromException instead of throwing